src/components/MainHead.astro

```diff
@@ -10,12 +10,26 @@ const {
   title = "Frontend Hire: Free Portfolio Template",
   description = "A free template that you can use to build your portfolio in minutes.",
 } = Astro.props;
+const urlStr = Astro.url.toString();
+const urlWithTrailingSlash = urlStr.endsWith("/") ? urlStr : urlStr + "/";
+const opengraphImageUrl = new URL(`./opengraph-image`, urlWithTrailingSlash);
 ---
 
 <meta charset='UTF-8' />
 <meta name='description' property='og:description' content={description} />
 <meta name='viewport' content='width=device-width' />
 <meta name='generator' content={Astro.generator} />
+<meta property='og:title' content={title} />
+<meta property='og:image' content={opengraphImageUrl.toString()} />
+<meta property='og:image:type' content='image/png' />
+<meta property='og:image:width' content='1200' />
+<meta property='og:image:height' content='630' />
+<meta name='twitter:card' content='summary_large_image' />
+<meta name='twitter:site' content='@thisisyhr' />
+<meta name='twitter:creator' content='@thisisyhr' />
```

Comment:
I do not like this being hard-coded. 

`site-config.ts` can be used to grab this stuff.

Reply:
Ok I will do this change